packages/core/src/connectors/injected.ts

```diff
@@ -34,6 +34,9 @@ export type InjectedParameters = {
   target?: TargetId | Target | (() => Target | undefined) | undefined
 }
 
+// Regex of wallets/providers that can accurately simulate contract calls & display contract revert reasons.
+const skipSimulateContractRegex = /(rabby|trustwallet)/
```

Comment:
Sadly, from my testing of the top wallets, only Rabby and Trust can accurately simulate contract writes.